src/main/java/nextstep/subway/applicaion/JGraphPathFinder.java

```diff
@@ -0,0 +1,44 @@
+package nextstep.subway.applicaion;
+
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.Path;
+import nextstep.subway.domain.Station;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.springframework.stereotype.Component;
+
+import java.util.Collection;
+import java.util.List;
+
+@Component
+public class JGraphPathFinder implements PathFinder {
+    @Override
+    public Path find(List<Line> lines, Station source, Station target) {
+        WeightedMultigraph<Long, DefaultWeightedEdge> subwayGraph = makeSubwayGraph(lines);
+        DijkstraShortestPath<Long, DefaultWeightedEdge> dijkstraShortestPath = new DijkstraShortestPath<>(subwayGraph);
+        int distance = (int) dijkstraShortestPath.getPathWeight(source.getId(), target.getId());
+        List<Long> paths = dijkstraShortestPath.getPath(source.getId(), target.getId()).getVertexList();
+        return new Path(paths, distance);
+    }
+
+    private WeightedMultigraph<Long, DefaultWeightedEdge> makeSubwayGraph(List<Line> lines) {
+        WeightedMultigraph<Long, DefaultWeightedEdge> graph = new WeightedMultigraph<>(DefaultWeightedEdge.class);
```

Comment:
```suggestion
        WeightedMultigraph<Station, DefaultWeightedEdge> graph = new WeightedMultigraph<>(DefaultWeightedEdge.class);
```

이렇게 바꿔보면 어떨까요?
Long이 무엇을 의미하는지 파악하려면 코드를 분석해야하는데 클래스를 명시하면 무엇의 거리를 구하는지 파악하기 좋지 않을까요?

Reply:
Station이 파악하기 좋은 것 같네요! 변경해보겠습니다